processor/elasticprocessor/processor.go

```diff
@@ -4,44 +4,61 @@ import (
 	"context"
 	"strings"
 
-	"github.com/tommyers-elastic/opentelemetry-collector-contrib/processor/elasticprocessor/internal/hostmetrics"
-
+	"github.com/elastic/opentelemetry-lib/remappers/hostmetrics"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.opentelemetry.io/collector/processor"
-
 	"go.uber.org/zap"
 )
 
+// remapper interface defines the Remap method that should be implemented by different remappers
+type remapper interface {
+	Remap(pmetric.ScopeMetrics, pmetric.MetricSlice, pcommon.Resource)
+}
+
 type ElasticProcessor struct {
-	cfg     *Config
-	logger  *zap.Logger
-	storage map[string]any
+	cfg       *Config
+	logger    *zap.Logger
+	storage   map[string]any
+	remappers []remapper
 }
 
 func newProcessor(set processor.CreateSettings, cfg *Config) *ElasticProcessor {
-	return &ElasticProcessor{cfg: cfg, logger: set.Logger, storage: make(map[string]any)}
+	remappers := []remapper{
+		hostmetrics.NewRemapper(set.Logger, hostmetrics.WithSystemIntegrationDataset(true)),
+	}
```

Comment:
You can do an early check for the `AddSystemMetrics` flag and not define any remappers if it is disabled. This will allow the usage of the remappers in the code to be more readable:

```suggestion
	var remappers []remapper
	if cfg.AddSystemMetrics {
		remappers := []remapper{
			hostmetrics.NewRemapper(set.Logger, hostmetrics.WithSystemIntegrationDataset(true)),
		}
	}
```

Reply:
Agreed!